src/blueprint/into_schema.rs

```diff
@@ -48,9 +75,30 @@ fn to_type(def: &Definition) -> dynamic::Type {
             }
             Some(expr) => {
               let expr = expr.to_owned();
+              let def_name = def_name.clone();
+              let field_name = field_name.clone();
+              let res_cache = res_cache.clone();
               FieldFuture::new(async move {
                 let ctx = EvaluationContext::new(req_ctx, &ctx);
-                let const_value = expr.eval(&ctx).await?;
+                let mut hasher = DefaultHasher::new();
+                let state = &mut hasher;
+                def_name.hash(state);
+                field_name.hash(state);
+
+                if let Some(const_value) = ctx.graphql_ctx.parent_value.as_value() {
+                  hash_const_value(const_value, state)
+                }
+
+                let mut args_list: Vec<_> = ctx.graphql_ctx.args.iter().collect();
+                args_list.sort_by(|(key1, _), (key2, _)| key1.cmp(key2));
+                args_list.iter().for_each(|(key, value)| {
+                  key.hash(state);
+                  hash_const_value(value.as_value(), state);
+                });
+
+                let key = hasher.finish();
+                let const_value = res_cache.fetch(&ctx, &expr, key).await?;
+
```

Comment:
Move the logic to read from cache into a separate function `read_from_cache`

Reply:
created a method in `RequestContext` named `cache_read` for reading from cache since it holds the `cache` object now